web/src/banners.ts

```diff
@@ -18,9 +18,19 @@ export type AlertBanner = Banner & {
     process: string;
 };
 
-export function open(banner: Banner | AlertBanner, $banner_container: JQuery): void {
+export function open(
+    banner: Banner | AlertBanner,
+    $banner_container: JQuery,
+    remove_after?: number,
+): void {
     const banner_html = render_banner(banner);
     $banner_container.html(banner_html);
+
+    if (remove_after !== undefined) {
+        setTimeout(() => {
+            close($banner_container.find(".banner"));
+        }, remove_after);
+    }
```

Comment:
This logic looks erroneous to me. The `close` function will be called after some duration `remove_after`, but by that time other banners could have been appended to the same container. Using `$banner_container.find(".banner")` could then select the wrong banner.

Reply:
What we could do is to modify the `open` method to return a `$banner` JQuery object, and then define another `open_and_close` method which adds the `delay` parameter.